### PR TITLE
add automaxprocs to default GOMAXPROCS to be CPU limit aware

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes https://github.com/gohugoio/hugo/issues/10950

This implementation does NOT log unlike importing the root automaxprocs package in #10951 

`go test ./...` passes locally on this PR, I can reproduce the failures on #10951 related to undesired logging output.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -17,10 +17,14 @@ import (
...
 	"os"
 
+	"go.uber.org/automaxprocs/maxprocs"
+
 	"github.com/gohugoio/hugo/commands"
 )
 
 func main() {
+	// default GOMAXPROCS to be CPU limit aware, still respecting GOMAXPROCS env
+	maxprocs.Set()
```
Comment (bep) [inline main.go:27]: Could you 

1. Move this into commands.Execute so it actually gets executed during the integration tess.
2. Make the comment into a proper sentence starting with a capital D and ending with a period.
3. Adjust the commit message title in line with the guidelines in this repo (start out with a capital A)

**Review by bep — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (bep) [reply main.go:27]: Also, it would be good if you could rebase it against the recently pushed master.

Comment (BenTheElder) [reply main.go]: All three should be done.

**Comment by BenTheElder** [comment]
So far I can't repro the CI test failure locally with go1.20.4 on latest macOS. `go test -race ./... -tags extended` passes.

**Pull request merged**